src/Shader.cpp

```diff
@@ -21,7 +21,7 @@ using namespace glm;
 // file reading
 void getFileContents(const char* filename, vector<char>& buffer) {
   ifstream file(filename, ios_base::binary);
-  if (file) {
+  if (file.good()) {
```

Comment:
What does this bring?

Reply:
ifstream here is a reference type so **file** will never be null, so wouldn't evaluating it always be "true" and evaluating "good()" returns true if both the fail and is bad bits are false?

Reply:
Yes, this is evaluated, via its `operator bool()`.

See: https://www.cplusplus.com/reference/ios/ios/operator_bool/

Reply:
i just pushed a revert for this concern. Thanks for pointing it out.